### PR TITLE
Each and method spec clean up

### PR DESCRIPTION
**Pull request by nikasulo**
# Description
I cleaned up a bit of the enumerable method code and used the `.self` syntax in the `my_each` and `my_each_with_index` methods. I also updated the tests for my each so that it returns an empty array

# How it works
`.self` is called on the object itself and we use the size property of the object to iterate through it however many times like so:
`def my_each
    self.size.times {|index|
       yield(self[index])
    }`

### CODE REVIEW

**Comment by evaveskova** [comment]
The original code from enumerable.methods.rb is missing.

File: spec/enumerable-spec.rb [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+describe '#my_each' do
+  let(:array) { [1,7,3,4,7] }
+
+  it 'returns the array' do
+    expect do
+      array.my_each {|a| print a}
+    end.to output("17347").to_stdout
+  end
+
+it 'returns an empty array' do
```
Comment (nikasulo) [inline spec/enumerable-spec.rb:12]: @evaveskova this is the only thing that will change if you merge
